internal/generator/common/utils.go

```diff
@@ -357,3 +359,50 @@ func CtxClosed(ctx context.Context) bool {
 		return false
 	}
 }
+
+func ExtractValuesFromTemplate(template string) []string {
+	re := regexp.MustCompile(`{{.*?\.([^\s|}]+).*?}}`)
+	matches := re.FindAllStringSubmatch(template, -1)
+
+	values := make([]string, 0, len(matches))
+
+	for _, match := range matches {
+		values = append(values, match[1])
+	}
+
+	return values
+}
```

Comment:
We are currently using golang templates, which prohibit accessing fields that contain characters other than these characters [A-Za-z0-9_].

However, we support the generation of columns with these symbols. 

For example, the column name `specific-name` is considered valid, but template `{{ .specific-name }}` is not.

Golang templates provide the ability to access a field like this
`{{ index . "specific-name" }}`

I propose to support such an appeal.

```suggestion
func ExtractValuesFromTemplate(template string) []string {
	// regexp for templates like {{ .name }}
	reField := regexp.MustCompile(`{{\s*(?:\w+\s+)?\.([^\s|}]+).*?}}`)
	matchesField := reField.FindAllStringSubmatch(template, -1)

	// regexp for templates like {{ index . "name-with-specific-symbols" }}
	reMapKey := regexp.MustCompile(`{{\s*index\s+\.\s+"([^"]+)".*?}}`)
	matchesMapKeys := reMapKey.FindAllStringSubmatch(template, -1)

	values := make([]string, 0, max(len(matchesField), len(matchesMapKeys)))

	for _, match := range matchesField {
		values = append(values, match[1])
	}
	for _, match := range matchesMapKeys {
		values = append(values, match[1])
	}

	return values
}
```

Reply:
Updated function